### DISCUSSION TITLE
Sorting on a belongsToThrough relation

### DISCUSSION THREAD

**Discussion by randohinn** [spatie/laravel-query-builder#847] [Q&A]
I have a relation in my model, defined as

```php
public function teacher() {
        return $this->belongsToThrough(User::class, Teacher::class);
}
```

So I get the user that is the teacher for this Camp model instance (the teachers table itself is for managing rights). 

What I am trying to achieve is to be be able to sort the user's `name` field. Tried this, no luck, getting a `SQLSTATE[42S22]: Column not found: 1054 Unknown column 'teacher.user.name' in 'order clause' ` instead.

```php
 $camps = QueryBuilder::for(Camp::class)
            ->allowedSorts(['id','teacher.user.name','days.0.date'])
            ->join('teachers', 'teachers.id', '=', 'camps.teacher_id')
            ->join('locations', 'locations.id', '=', 'camps.location_id')
            ->with(['teacher','days','location'])
            ->select('camps.*')
            ->get();
````

**Comment by lebadapetru** [comment]
have you found a way to sort using a relation field?
